docs/site/Binding.md

```diff
@@ -148,6 +148,127 @@ binding.inScope(BindingScope.SINGLETON);
+`to()`. For example:
+
+```ts
+ctx.bind('my-name').to('John Smith');
+```
+
+The `my-name` binding will always resolve to `'John Smith'`.
+
+The binding scope will impact values provided by `toDynamicValue`, `toClass`,
+and `toProvider`.
+
+Let's say we need to have a binding that gives us the current date.
+
+```ts
+ctx.bind('current-date').toDynamicValue(() => new Date());
+const d1 = ctx.getSync('current-date');
+const d2 = ctx.getSync('current-date');
+```
+
+By default, the binding scope is `TRANSIENT`. In the code above, `d1` and `d2`
+are resolved by calling `new Date()` for each `getSync('current-date')`. Two
+different dates are assigned to `d1` and `d2` to reflect the corresponding date
+for each resolution.
+
+Now you can guess the code snippet below will produce the same date for `d1` and
+`d2`, which is not desirable.
+
+```ts
+ctx
+  .bind('current-date')
+  .toDynamicValue(() => new Date())
+  .inScope(BindingScope.SINGLETON);
+const d1 = ctx.getSync<Date>('current-date');
+const d2 = ctx.getSync<Date>('current-date');
+```
+
+The `SINGLETON` scope is useful for some use cases, such as:
+
+1.  Share states across multiple consumers of the binding
+
+    ```ts
+    export class GlobalCounter {
+      public count = 0;
+    }
+
+    ctx
+      .bind('global-counter')
+      .toClass(GlobalCounter)
+      .inScope(BindingScope.SINGLETON);
+    const c1: GlobalCounter = await ctx.get('global-counter');
+    c1.count++; // c1.count is now 1
+    const c2: GlobalCounter = await ctx.get('global-counter');
+    // c2 is the same instance as c1
+    // c2.count is 1 too
+    ```
+
+2.  Save creation of multiple instances if each instance does not need to hold
+    or access different states
+
+    For example, the following `GreetingController` implementation does not
+    access any information beyond the method parameters and it is a good
+    candidate to use `SINGLETON` so that only one instance is created within the
+    application context and it's shared by all requests.
+
+    ```ts
+    @bind({scope: BindingScope.SINGLETON})
+    export class GreetingController {
+      greet(name: string) {
+        return `Hello, ${name}`;
+      }
+    }
+
+    const binding = ctx.add(createBindingFromClass(GreetingController));
+    const c1 = ctx.getSync(binding.key);
+    const c2 = ctx.getSync(binding.key);
+    // c2 is the same instance as c1
+    ```
+
+**Rule of thumb**: Use `TRANSIENT` as the safe default and choose `SINGLETON` if
+you want to share the same instance for all consumers.
+
+Let's look at another use case that we need to access the information from the
+current request, such as http url or logged in user:
+
+```ts
+export class GreetingCurrentUserController {
+  @inject(SecurityBindings.USER)
+  private currentUserProfile: UserProfile;
+
+  greet() {
+    return `Hello, ${this.currentUserProfile.name}`;
+  }
+}
+```
+
+The `GreetingCurrentUserController` has to use `TRANSIENT` scope so that a new
```

Comment:
where is the `@bind({scope: BindingScope.TRANSIENT}) in this example?

Reply:
By default, `GreetingCurrentUserController` is bound in `TRANSIENT` scope. Please note `@bind` is a hint for `createBindingFromClass` to set up the scope but it is not required. For example:

```ts
ctx.bind('current-date').toDynamicValue(() => new Date()).inScope(BindingScope.TRANSIENT);
```

Is the same as:
```ts
ctx.bind('current-date').toDynamicValue(() => new Date());
```

Reply:
See https://loopback.io/doc/en/lb4/Binding.html#configure-binding-attributes-for-a-class